{"name": "test_patch_user", "status": "failed", "statusDetails": {"message": "AssertionError: Expected status code 200, but got 404", "trace": "api_setup = <utils.service_api.ServiceAPINew.ServiceAPI object at 0x000001B9ADAE55E0>, base_url = 'https://companydev.jobeze.com/backend', created_user_id = 2128\n\n    def test_patch_user(api_setup, base_url, created_user_id):\n        \"\"\"Test the PATCH request for a specific user ID.\"\"\"\n        patch_url = f\"{base_url}/users/{created_user_id}\"\n        payload_patch = {\"firstName\": \"UpdatedFirstName\"}\n        response_patch = api_setup.patch_service_response(patch_url, payload_patch)\n>       api_setup.validate_response(response_patch, expected_status=200)\n\ntestsuites\\api\\test_service_api.py:66: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <utils.service_api.ServiceAPINew.ServiceAPI object at 0x000001B9ADAE55E0>, response = <Response [404]>, expected_status = 200\n\n    @allure.step(\"Validating API response\")\n    def validate_response(self, response, expected_status=200):\n        \"\"\"Validate the API response with the expected status.\"\"\"\n        assert response is not None, \"No response received from the API\"\n>       assert response.status_code == expected_status, \\\n            f\"Expected status code {expected_status}, but got {response.status_code}\"\nE       AssertionError: Expected status code 200, but got 404\n\nutils\\service_api\\ServiceAPINew.py:89: AssertionError"}, "description": "Test the PATCH request for a specific user ID.", "steps": [{"name": "Sending PATCH request to API", "status": "passed", "parameters": [{"name": "api_url", "value": "'https://companydev.jobeze.com/backend/users/2128'"}, {"name": "payload", "value": "{'firstName': 'UpdatedFirstName'}"}], "start": 1746425211995, "stop": 1746425212215}, {"name": "Validating API response", "status": "failed", "statusDetails": {"message": "AssertionError: Expected status code 200, but got 404\n", "trace": "  File \"C:\\Internal\\automation-framework-master\\venv\\Lib\\site-packages\\allure_commons\\_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"C:\\Internal\\automation-framework-master\\utils\\service_api\\ServiceAPINew.py\", line 89, in validate_response\n    assert response.status_code == expected_status, \\\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n"}, "parameters": [{"name": "response", "value": "<Response [404]>"}, {"name": "expected_status", "value": "200"}], "start": 1746425212215, "stop": 1746425212215}], "attachments": [{"name": "log", "source": "b00289dd-be21-4f33-b51e-3633c0a0eca6-attachment.txt", "type": "text/plain"}], "start": 1746425211995, "stop": 1746425212216, "uuid": "3c90190d-e422-4afe-a231-435b95a66093", "historyId": "ba5c98e5ad40f6e362a9c5efe43d9378", "testCaseId": "ba5c98e5ad40f6e362a9c5efe43d9378", "fullName": "testsuites.api.test_service_api#test_patch_user", "labels": [{"name": "parentSuite", "value": "testsuites.api"}, {"name": "suite", "value": "test_service_api"}, {"name": "host", "value": "DESKTOP-BR6BRT3"}, {"name": "thread", "value": "23956-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "testsuites.api.test_service_api"}]}